### PR TITLE
Add support for joins

### PR DESCRIPTION
**Pull request by mcbrewster**
Adds support for joins

### CODE REVIEW

File: src/parser/druidsql.pegjs [hunk 1]
```diff
@@ -173,6 +185,21 @@ FromClause = fromKeyword:FromToken postFrom:_ tableHead:(Alias/SqlRef) tableTail
...
 
+JoinClause = joinType:JoinType postJoinType:_ joinKeyword:JoinToken postJoinKeyword:_? table:(Alias/SqlRef) postJoinTable:_  onKeyword:OnToken postOn:_? onExpression:Expression
+{
+  return {
+    joinType: joinType,
+    postJoinType: postJoinType,
+    joinKeyword: joinKeyword,
+    postJoinKeyword: postJoinKeyword,
+    table: table,
+    postJoinTable: postJoinTable,
```
Comment (vogievetsky) [inline src/parser/druidsql.pegjs]: Could you please suffix everything that is a space with `Spacing`

File: src/sql/sql-query/sql-query.spec.ts [hunk 2]
```diff
@@ -17,2470 +17,3278 @@ import { FUNCTIONS } from '../../test-utils';
...
+//
+//     expect(parser(sql)).toMatchInlineSnapshot(`
+//       SqlQuery {
+//         "explainKeyword": "",
+//         "fromKeyword": "from",
+//         "groupByExpression": undefined,
+//         "groupByExpressionSeparators": undefined,
+//         "groupByKeyword": undefined,
+//         "havingExpression": undefined,
+//         "havingKeyword": undefined,
```
Comment (vogievetsky) [inline src/sql/sql-query/sql-query.spec.ts]: no commented out code please

File: src/parser/druidsql.pegjs [hunk 3]
```diff
@@ -637,6 +664,13 @@ Decorator =
...
+JoinType =
+    'LEFT'i
+    /'RIGHT'i
+    /'INNER'i
+    /$('FULL'i _ 'OUTER'i)
+    /'FULL'i
+
 OrToken = 'OR'i
 AndToken = 'AND'i
 NotToken = 'NOT'i
```
Comment (vogievetsky) [inline src/parser/druidsql.pegjs]: Could you please try to parse/stringily something like `Select notingham from table`

You might need a nice little `!IdentifierPart` on all of these. See: https://github.com/implydata/plywood/blob/master/src/expressions/plyql.pegjs#L1053

Comment (mcbrewster) [reply src/parser/druidsql.pegjs]: `Select notingham from table` works fine but i will add a simple test anyway. The reason this isn't necessary is that in ply wood white space is always optional but in DQT is always enforced which will prevent the rule from matching on a string with out the need for !IdentifierPart.

File: src/parser/druidsql.pegjs (hunk 1 above)
Comment (mcbrewster) [reply src/parser/druidsql.pegjs]: JoinClause is not a class so all all the spacing is eventually added to the inner spacing of the SqlQueryObject, I can add it for just the rule though.

File: src/parser/druidsql.pegjs (hunk 3 above)
Comment (vogievetsky) [reply src/parser/druidsql.pegjs]: cool cool cool

**Comment by vogievetsky** [comment]
gr 🎱

**Pull request merged**